pkg/datasources/tables_acceptance_test.go

```diff
@@ -13,11 +13,13 @@ func TestAccTables(t *testing.T) {
 	databaseName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
 	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
 	tableName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	stageName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	externalTableName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
 	resource.ParallelTest(t, resource.TestCase{
 		Providers: providers(),
 		Steps: []resource.TestStep{
 			{
-				Config: tables(databaseName, schemaName, tableName),
+				Config: tables(databaseName, schemaName, tableName, stageName, externalTableName),
```

Comment:
I created external table in this test.
If external table is counted, "1" of resource.TestCheckResourceAttr("data.snowflake_tables.t", "tables.#", "1") become "2". If not, it becomes "1"